crowbar_framework/app/models/node_object.rb

```diff
@@ -835,6 +841,20 @@ def _remove_elements_from_node(old, new, from_node)
     @attrs_last_saved = @role.default_attributes.deep_dup
 
     Rails.logger.debug("Done saving node: #{@node.name} - #{crowbar_revision}")
+
+    unless @on_save_events.empty?
+      Rails.logger.debug("Notifying about saved node: #{@node.name}")
+
+      details = {
+        node: @node.name,
+        attributes: @on_save_events
+      }
+      Crowbar::EventDispatcher.trigger_hooks(:node_changed, details)
+
+      @on_save_events = Set.new
```

Comment:
The node object is not shared among puma threads, so it should be fine. When we create threads ourselves, we should be careful of course and use proper locking mechanisms, but it's not specific to this change.